src/charm.py

```diff
@@ -709,11 +732,17 @@ def _generate_grafana_config(self) -> str:
         For now, this only creates database information, since everything else
         can be set in ENV variables, but leave for expansion later so we can
         hide auth secrets
+
+        The feature toggle accessTokenExpirationCheck is also set here. It's needed
+        for the oauth relation to provide refresh tokens.
         """
         configs = []
         if self.has_db:
             configs.append(self._generate_database_config())
 
+        if self.model.relations[OAUTH]:
```

Comment:
Let's keep it consistent
```suggestion
        if self.oauth.is_client_created():
```